src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -581,6 +585,9 @@ public String getDatabaseProductVersion() throws SQLException {
             return serverVersion;
         }
         serverVersion = conn.getServerVersion();
+        String[] versionParts = serverVersion.split("\\.");
+        serverMajorVersion = parseInt(versionParts[0]);
```

Comment:
If you don't check the length, this could throw an exception. Not sure how much we care.